src/components/VideoPlayer/VideoPlayerControls/ProgressBar/index.js

```diff
@@ -0,0 +1,69 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {Gesture, GestureDetector} from 'react-native-gesture-handler';
+import Animated, {runOnJS, useAnimatedStyle, useSharedValue} from 'react-native-reanimated';
+import {usePlaybackContext} from '@components/VideoPlayerContexts/PlaybackContext';
+import useThemeStyles from '@hooks/useThemeStyles';
+
+const propTypes = {
+    duration: PropTypes.number.isRequired,
+
+    position: PropTypes.number.isRequired,
+
+    seekPosition: PropTypes.func.isRequired,
+
+    togglePlayCurrentVideo: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
+
+function ProgressBar({togglePlayCurrentVideo, duration, position, seekPosition}) {
+    const styles = useThemeStyles();
+    const {pauseVideo} = usePlaybackContext();
+    const [sliderWidth, setSliderWidth] = useState(1);
+    const progressWidth = useSharedValue(0);
+
+    const onSliderLayout = (e) => {
+        setSliderWidth(e.nativeEvent.layout.width);
+    };
+
+    const pan = Gesture.Pan()
+        .onBegin(() => {
+            runOnJS(pauseVideo)();
+        })
+        .onChange((event) => {
+            progressWidth.value = (event.x / sliderWidth) * 100;
+            runOnJS(seekPosition)((event.x / sliderWidth) * duration);
+        })
+        .onEnd(() => {
+            runOnJS(togglePlayCurrentVideo)();
```

Comment:
Another odd consequence of the `pause` in the `onBegin()` is that if a video is originally paused and then you change the position on the progress bar, it starts playing the video here, which also feels unexpected.

Reply:
The behaviour that the video is paused when user interacts with the progress bar is common, but I need to figure out how to remember previous state of the playback. Will focus on it tomorrow.

Reply:
It's fixed